lxd/containers.go

```diff
@@ -1719,6 +1737,16 @@ func (c *lxdContainer) setupMacAddresses(d *Daemon) error {
 			}
 		}
 
+		/*
+		 * propagate the name of the device down for nics if the user
+		 * didn't specify one. This is probably what they wanted
+		 * anyway, and lets us match against network names once they go
+		 * into lxc, e.g. to pull out the veth name for `lxc info`.
+		 */
+		if _, ok := d["name"]; !ok {
+			d["name"] = name
+		}
+
```

Comment:
On Mon, Jun 15, 2015 at 08:17:52AM -0700, Stéphane Graber wrote:
> > @@ -1719,6 +1737,16 @@ func (c *lxdContainer) setupMacAddresses(d *Daemon) error {
> >  			}
> >  		}
> >  
> > +		/*
> > +		 * propagate the name of the device down for nics if the user
> > +		 * didn't specify one. This is probably what they wanted
> > +		 * anyway, and lets us match against network names once they go
> > +		 * into lxc, e.g. to pull out the veth name for `lxc info`.
> > +		 */
> > +		if _, ok := d["name"]; !ok {
> > +			d["name"] = name
> > +		}
> > +
> 
> Network in the container.

Why is that? You're not allowed to have a device with the same name
inside a net ns?